addon/utils/handle-query-params.js

```diff
@@ -0,0 +1,15 @@
+export default function(params) {
+  let possibleQueryParams = params[params.length - 1];
+
+  if (possibleQueryParams && possibleQueryParams.isQueryParams) {
+
+    // ensure to handle potentially frozen arrays
+    params = params.slice();
```

Comment:
FWIW, we are avoiding the slice in the case when the last param is not a query param.